src/common/transformations/tests/common_optimizations/convert_nms_gather_path_to_unsigned_test.cpp

```diff
@@ -205,3 +205,62 @@ TEST(TransformationTests, test_convert_to_unsigned_nms_gather_3) {
     ASSERT_NO_THROW(check_rt_info(f));
     ASSERT_EQ(count_ops_of_type<ov::op::v0::Convert>(f), 0);
 }
+
+TEST(TransformationTests, test_convert_to_unsigned_nms_gather_with_if_condition) {
+    auto boxes = make_shared<opset8::Parameter>(element::f32, PartialShape{1, -1, 4});
+    auto scores = make_shared<opset8::Parameter>(element::f32, PartialShape{1, 1, -1});
+    auto nms = make_shared<opset8::NonMaxSuppression>(boxes, scores);
+
+    auto gather = make_shared<opset8::Gather>(nms->output(0),
+                                              opset8::Constant::create(element::i32, Shape{1}, {2}),
+                                              opset8::Constant::create(element::i32, Shape{1}, {0}));
+
+    auto shape_of = make_shared<opset8::ShapeOf>(gather);
+    auto gather_shape = make_shared<opset8::Gather>(shape_of,
+                                                    opset8::Constant::create(element::i32, Shape{1}, {0}),
+                                                    opset8::Constant::create(element::i32, Shape{1}, {0}));
+    auto equal = make_shared<opset8::Equal>(gather_shape, opset8::Constant::create(element::i64, Shape{1}, {1}));
+    auto if_op = make_shared<opset8::If>(equal);
+
+    auto input_then = std::make_shared<opset8::Parameter>(element::i32, PartialShape{-1, 1});
```

Comment:
Loop operation can contain any operations inside "then/else" bodies
So in some cases, we would not want to propagate NMSSelectedIndices key through IF operation, e.g. if "then/else" bodies contain operations which are not from this list
 
<img width="214" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/21181928/9eba85c9-b6dc-4a89-be06-65e1ffe176a4">

Better solution is to re-use PropagateNMSPath for If operation bodies to propogate the keys correctly.